### PR TITLE
add Store#hasModule(path) API

### PR DESCRIPTION
**Pull request by FranckFreiburger**

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -179,6 +179,16 @@ export class Store {
...
   }
 
+  hasModule (path) {
+    if (typeof path === 'string') path = [path]
+
+    if (process.env.NODE_ENV !== 'production') {
+      assert(Array.isArray(path), `module path must be a string or an Array.`)
+    }
+
+    this._modules.isRegistered(path)
```
Comment (ktsn) [inline src/store.js]: We must return the result of this expression.

File: src/module/module-collection.js [hunk 2]
```diff
@@ -53,6 +53,14 @@ export default class ModuleCollection {
 
     parent.removeChild(key)
   }
+
+  isRegistered (path) {
+    const parent = this.get(path.slice(0, -1))
```
Comment (ktsn) [inline src/module/module-collection.js]: I realized this line can throw an error if the parent module does not exist.
To handle this, we also need to add existance check in `get` method.

Comment (FranckFreiburger) [inline src/module/module-collection.js]: yes,
```
store.registerModule('hi' ...
expect(store.hasModule('hi')).toBe(true)
expect(store.hasModule(['hi','big'])).toBe(false)
expect(store.hasModule(['hi','big', 'world'])).toBe(false) // THROW
expect(store.hasModule(['bonjour'])).toBe(false)

```

Comment (FranckFreiburger) [inline src/module/module-collection.js]: `store.registerModule(['aze', 'rty'], ... // THROW`

Comment (FranckFreiburger) [inline src/module/module-collection.js]: maybe this behavior is expected, maybe it's a documentation issue ?

Comment (ktsn) [inline src/module/module-collection.js]: But the purpose of this feature is to check existance of module, isn't it? I think the `hasModule` should return `false` in that case.

Comment (FranckFreiburger) [inline src/module/module-collection.js]: yes, this feature is to check existence of module, but maybe not the existence of sub-module.  
It's quite like `mkdir` and `mkdirp`

Comment (FranckFreiburger) [inline src/module/module-collection.js]: I propose to add `ModuleCollection#has(path)`

Comment (ktsn) [inline src/module/module-collection.js]: I'm not sure why this method is semantically related with `mkdir`. They are used on different platform (Web/CLI) and also are different kind of task (`hasModule` *checks* a module while `mkdir` *creates* a directory).

If the parent module does not exist, it's clear that its child module also does not exist. So why `hasModule` need to throw in that case? 🤔

**Comment by ktsn** [comment]
* Can we add **a new test case** for `hasModule`? Adding a test code to non-relevant test case would problematic since it would make harder to detect a real problem when the test cases fail.
* Looks like CI is failed due to linting issue. Please check/fix it by running `npm test` on your terminal.

File: src/module/module-collection.js (hunk 2 above)
Comment (FranckFreiburger) [inline src/module/module-collection.js]: Trust me, I am not trying to create modules with mkdir :smiley:
I just wanted to say that the responsibility lies with the caller to ensure the correctness of the arguments.

**Comment by FranckFreiburger** [comment]
At this point, `hasModule()` behaves like `register()` and `unregister()` in term of error management, it throws if parent module does not exist.
I will propose a new PR for existence check in get method and update these 3 methods.

**Comment by anteriovieira** [comment]
What prevents it from being merged?

**Comment by trainiac** [comment]
@ktsn Can this be merged? Or does this PR need more work?

**Comment by mlanin** [comment]
@ktsn a year has passed. Can we merge this already? Please :)

**Comment by wa3l** [comment]
@yyx990803 any timeline on merging this?

**Comment by mlavina** [comment]
@ktsn @yyx990803 This would also be really helpful to my team can we get this merged in and released?

**Comment by JasonLandbridge** [comment]
2020 now, this is such a small pull request. Just merge it already!

**Comment by bhaskar-nair2** [comment]
Waiting for this to be available in master...

**Pull request merged**